addons/sale_project/models/project.py

```diff
@@ -54,6 +54,27 @@ class ProjectTask(models.Model):
     invoice_count = fields.Integer("Number of invoices", related='sale_order_id.invoice_count')
     task_to_invoice = fields.Boolean("To invoice", compute='_compute_task_to_invoice', search='_search_task_to_invoice', groups='sales_team.group_sale_salesman_all_leads')
 
+    @api.depends('sale_line_id', 'project_id', 'commercial_partner_id')
+    def _compute_sale_order_id(self):
+        for task in self:
+            sale_order_id = task.sale_order_id or self.env["sale.order"]
+            if task.sale_line_id:
+                sale_order_id = task.sale_line_id.sudo().order_id
+            elif task.project_id.sale_order_id:
+                sale_order_id = task.project_id.sale_order_id
+            if task.commercial_partner_id != sale_order_id.partner_id.commercial_partner_id:
+                sale_order_id = False
+            if sale_order_id and not task.partner_id:
+                task.partner_id = sale_order_id.partner_id
+            task.sale_order_id = sale_order_id
+
+    @api.depends('project_id.sale_line_id.order_partner_id')
+    def _compute_partner_id(self):
+        for task in self:
+            if not task.partner_id:
+                task.partner_id = task.project_id.sale_line_id.order_partner_id
+        super()._compute_partner_id()
```

Comment:
This method has been removed in this [PR](https://github.com/odoo/odoo/pull/51471/files#diff-2f1670bdaecbb290806818cc80aff338e2f956d89f3a71f6c2ed1b9805795fd3L57-L62), normally you do not need this method anymore 🙂